src/libsyntax_ext/source_util.rs

```diff
@@ -153,7 +153,7 @@ pub fn expand_include_bytes(cx: &mut ExtCtxt<'_>, sp: Span, tts: &[tokenstream::
                 },
                 Err(e) => (String::new(), e.into_bytes()),
             };
-            cx.source_map().new_source_file(file.into(), contents);
+            cx.source_map().new_source_file(&cx.parse_sess.span_diagnostic, file.into(), contents);
```

Comment:
Ha.
`include_bytes!` certainly shouldn't do the normalization.
I didn't check all the regressions, but the ones I checked are caused by this.

Reply:
(It also shouldn't remove BOM, so if it did, it was a bug.)

Reply:
I *think* this is actually a bug, see https://github.com/rust-lang/rust/pull/54517/files

The TL;DR is that we want to add *something* to source_map to track dependencies (so adding "" is OK), **but** if we simulateneously include file as `include_bytes` and as a proper module, stuff will blow up because the module will see empty string. 

It centrally feels like "add file to source map to track deps" is a smell, at least because we essentially store the file twice (as `Vec<u8>` and as a `String`). Can we directly register file as a dependency?

Reply:
Ah, I see, so the byte literal contains the original data and the copy processed by `new_source_file` (which reports the "bare `\r`" errors in process) is thrown away.

>Can we directly register file as a dependency?

Probably?
I'm not familiar with that code but don't see why `parse_sess` couldn't e.g. contain a list of files to add to deps in addition to the source map files.

Reply:
Marking as "waiting-on-author" for investigation.
The `include_binary!` files either need to skip all the error reporting from `new_source_file`, or better extract the parts of `new_source_file` that are relevant to binary files (registering as dependency? maybe something else?) into some new `new_binary_file`.

Reply:
Another issue here is that we literally do `std::fs::read` here, dodging the `FileLoader` infra which doesn't have allowance for binary files at all...